controllers/vspherepolicy/policyevaluation/policyevaluation_controller_test.go

```diff
@@ -0,0 +1,2427 @@
+					BeforeEach(func() {
+						obj.Spec.Image = &vspherepolv1.PolicyEvaluationImageSpec{
+							Labels: map[string]string{
+								"version": "1.0",
+								"arch":    "amd64",
+								"os":      "ubuntu",
+							},
+						}
+					})
+
+					It("should match the policy", func() {
+						req := ctrl.Request{
+							NamespacedName: types.NamespacedName{
+								Name:      obj.Name,
+								Namespace: obj.Namespace,
+							},
+						}
+
+						result, err := reconciler.Reconcile(ctx, req)
+						Expect(err).ToNot(HaveOccurred())
+						Expect(result).To(Equal(ctrl.Result{}))
+
+						var updated vspherepolv1.PolicyEvaluation
+						Expect(client.Get(ctx, ctrlclient.ObjectKeyFromObject(obj), &updated)).To(Succeed())
+						Expect(updated.Status.Policies).To(HaveLen(1))
+						Expect(updated.Status.Policies[0].Name).To(Equal(computePolicy.Name))
+						Expect(updated.Status.Policies[0].APIVersion).To(Equal(vspherepolv1.GroupVersion.String()))
+						Expect(updated.Status.Policies[0].Kind).To(Equal("ComputePolicy"))
+						Expect(updated.Status.Policies[0].Generation).To(Equal(computePolicy.Generation))
+					})
+				})
+
+				Context("when PolicyEvaluation has non-matching image labels", func() {
+					BeforeEach(func() {
+						obj.Spec.Image = &vspherepolv1.PolicyEvaluationImageSpec{
+							Labels: map[string]string{
+								"version": "2.0",
+								"arch":    "arm64",
+							},
+						}
+					})
+
+					It("should not match the policy", func() {
+						req := ctrl.Request{
+							NamespacedName: types.NamespacedName{
+								Name:      obj.Name,
+								Namespace: obj.Namespace,
+							},
+						}
+
+						result, err := reconciler.Reconcile(ctx, req)
+						Expect(err).ToNot(HaveOccurred())
+						Expect(result).To(Equal(ctrl.Result{}))
+
+						var updated vspherepolv1.PolicyEvaluation
+						Expect(client.Get(ctx, ctrlclient.ObjectKeyFromObject(obj), &updated)).To(Succeed())
+						Expect(updated.Status.Policies).To(BeEmpty())
+					})
+				})
+			})
+
+			Context("with explicit policies", func() {
+				var computePolicy *vspherepolv1.ComputePolicy
+
+				BeforeEach(func() {
+					computePolicy = &vspherepolv1.ComputePolicy{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "explicit-compute-policy",
+							Namespace: namespace,
+						},
+						Spec: vspherepolv1.ComputePolicySpec{
+							EnforcementMode: vspherepolv1.PolicyEnforcementModeMandatory,
+							Match: &vspherepolv1.MatchSpec{
+								Workload: &vspherepolv1.MatchWorkloadSpec{
+									Labels: []metav1.LabelSelectorRequirement{
+										{
+											Key:      "env",
+											Operator: metav1.LabelSelectorOpIn,
+											Values:   []string{"dev"}, // Won't match our test object
+										},
+									},
+								},
+							},
+						},
+					}
+					withObjs = append(withObjs, computePolicy)
+
+					obj.Spec.Policies = []vspherepolv1.LocalObjectRef{
+						{
+							APIVersion: "vsphere.policy.vmware.com/v1alpha1",
```

Comment:
Nit: could be replaced with `vspherepolv1.GroupVersion.String()` so you don't have to change it in the future